### PR TITLE
Fix namespace bug

### PR DESCRIPTION
**Pull request by BenFordTytherington**
Putting Namespace attrs on a type in RustQt or the block itself doesn't compile, with a big string of cpp errors.

I've tried putting namespace back into the rust generate fn, along with a host of other options, such as putting the namespace on the method block as well.

However, putting namespace on the bridge works fine, and everything works properly.

### CODE REVIEW

File: crates/cxx-qt-gen/src/generator/rust/method.rs [hunk 1]
```diff
@@ -52,6 +59,7 @@ pub fn generate_rust_methods(
                     // CXX ends up generating the source, then we generate the matching header.
                     #[cxx_name = #wrapper_ident_cpp]
                     // Namespace is not needed here
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/rust/method.rs:61]: turns out it is when the namespace is on the `type MyObject` and not on the cxx_qt::bridge

File: crates/cxx-qt-gen/src/generator/naming/qobject.rs [hunk 2]
```diff
@@ -62,6 +63,15 @@ impl QObjectNames {
...
         )
     }
+
+    pub fn namespace_tokens(&self) -> TokenStream {
+        let namespace = self.name.namespace();
+        if namespace.is_none() {
+            quote! {}
+        } else {
+            quote! { #[namespace = #namespace ] }
+        }
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/naming/qobject.rs:73]: could be
```rust
if let Some(namespace) = self.name.namespace() {
    quote! { #[namespace = #namespace] }
else {
    quote! {}
}
```

File: crates/cxx-qt-gen/src/generator/rust/qobject.rs [hunk 3]
```diff
@@ -160,6 +160,7 @@ fn generate_qobject_definitions(
             },
             quote! {
                 extern "Rust" {
+                    #namespace // Needed for namespacing QObjects
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/rust/qobject.rs:163]: can the place where `let namespace` is done here use your new helper too ?

File: crates/cxx-qt-gen/src/generator/naming/qobject.rs [hunk 4]
```diff
@@ -62,6 +63,15 @@ impl QObjectNames {
             ident = self.name.cxx_unqualified().to_case(Case::Snake)
         )
     }
+
+    pub fn namespace_tokens(&self) -> TokenStream {
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/naming/qobject.rs:67]: doc line would be good

File: crates/cxx-qt-gen/src/generator/rust/property/getter.rs [hunk 5]
```diff
@@ -34,11 +34,13 @@ pub fn generate(
...
         let qualified_impl = type_names.rust_qualified(cpp_class_name_rust)?;
 
+        let cxx_namespace = qobject_idents.namespace_tokens();
+
         Ok(Some(RustFragmentPair {
             cxx_bridge: vec![quote! {
                 extern "Rust" {
                     #[cxx_name = #getter_wrapper_cpp]
-                    // Namespace is not needed here
+                    #cxx_namespace // Needed so QObjects can have a namespace
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/rust/property/getter.rs:43]: maybe change the comment to something like the following
```
// Needed for QObjects that have a namespace on their type item or extern block
//
// Note that a namespace in the cxx_qt::bridge would be applied to all child items
// but for other cases there is no parent so we need to be explicit
#cxx_namespace
```

Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/src/generator/rust/property/getter.rs:43]: (same for the others)

File: crates/cxx-qt-gen/src/generator/rust/qobject.rs [hunk 6]
```diff
@@ -160,6 +150,7 @@ fn generate_qobject_definitions(
             },
             quote! {
                 extern "Rust" {
+                    #namespace // Needed for namespacing QObjects individually, not just by namespacing entire bridge
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/rust/qobject.rs:153]: update the longer comment

Comment (BenFordTytherington) [reply crates/cxx-qt-gen/src/generator/rust/qobject.rs]: Done

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**